src/02-common-principles.md

```diff
@@ -383,6 +383,200 @@ In particular, if a BIDS dataset contains a `derivatives/` sub-directory,
+and the file will be found by navigating to the location specified in the BIDS URI
+relative to the downloaded data.
+
+### Refer to a file outside of a dataset but on the same host
+
+It is RECOMMENDED to refer to files **within** a dataset,
+or on a **remote** location as described in the sections above.
+However sometimes it may be convenient to refer to files that are outside of a given dataset but on the same host.
+BIDS URIs allow for specifying such locations,
+but such specifications are by definition not portable in that the BIDS URIs break when the host changes.
+When sharing a BIDS dataset, the dataset curator MUST make sure that all BIDS URIs are portable.
+
+To link for example to a derivative file that is stored in a `derivatives/`
+directory that is NOT nested in the raw BIDS directory:
+
+`bids:deriv3:/sub-01/anat/sub-01_desc-preproc_T1w.nii.gz`
+
+```json
+{
+    "DatasetLinks": {
+        "deriv3": "file://../some-local-path/derivative3"
+    }
+}
+```
+
+As shown in the example above, the path to `deriv3` contains the `..` syntax,
+which is a common way to refer to parent directories.
+In this way, arbitrary locations on the host can be referenced relative to the current dataset.
+
+Alternatively, one may specify an absolute location on the host using a
+forward slash directly after the `file://` URI scheme as such:
+`file:///`.
+
+On Unix-like operating systems (including MacOS) this refers to the system root.
+One may reference a dataset on some example user's Desktop as such:
+
+```json
+{
+    "DatasetLinks": {
+        "deriv3": "file:///home/example-user/Desktop/my-dataset"
+    }
+}
+```
+
+On the Windows operating system, `file:///` (note the `/` after the `file://` URI scheme)
+refers to a "virtual" system root that does not exist in practice.
+To reference a dataset on our example user's Windows Desktop, one may use:
+
+```json
+{
+    "DatasetLinks": {
+        "deriv3": "file:///C:/Users/example-user/Desktop/my-dataset"
+    }
+}
+```
+
+On a Windows machine, the location from this example would look like:
+`C:\Users\example-user\Desktop\my-dataset`
+
+However throughout BIDS, forward slashes `/` MUST be used as the path separator.
+Furthermore, the location `C:\` is specified in BIDS using the "virtual root"
+syntax as shown in the example: `/C:`
+
+### BIDS URI usage recommendations
+
+Dataset curators can flexibly use BIDS URIs to point to files in their BIDS datasets.
+However, when it comes to publicly sharing the data,
+BIDS makes two strong RECOMMENDATIONS:
+
+1.  Whenever possible, all data that is pointed to SHOULD be included in the BIDS dataset,
+    and BIDS URI pointers should be of the "local" form (that is, `bids::/absolute_path_within_dataset`).
+
+1.  In case shipping the data that is pointed to is impossible,
+    that data SHOULD be made available remotely with a DOI that allows an easy download from anywhere and by anyone.
```

Comment:
"data" -> "dataset" (see my previous comments on this)

Reply:
I agree with your previous comments and have addressed them in https://github.com/bids-standard/bids-specification/pull/820/commits/b82f7ad5eade8cf1cdf34e31ac1beec615232228

Here however, I think it doesn't apply. Let me explain in an example:

1. I have some file `BIDS-Specification-v1.6.0.pdf` that for some reason I cannot include in my BIDS dataset
2. This file is the "data" that I want to point to (let's ignore the fact that this is a PDF, and not some `.nii` file or similar)
3. The recommendation here tells me to upload this data to Zenodo, like here: https://zenodo.org/record/4710751/
4. The BIDS URI to that file would be: `bids:myds:/BIDS-Specification-v1.6.0.pdf`, where `dataset_description.DatasetLinks` has a field: `"myds": "doi:10.5281/zenodo.4710751"`
5. The "whole dataset" that I am asked to download is `"myds"`, under the specififed DOI
6. After downloading, I need to navigate to the path (relative to `"myds"`): `BIDS-Specification-v1.6.0.pdf` (it's at the root) to obtain the file

Of course I would be free to upload `BIDS-Specification-v1.6.0.pdf` as part of a larger dataset to Zenodo, and then in the BIDS URI refer to it with a different (or the same, if appropriate) path ... but that's not required.

Reply:
I believe the confusion remains of the same nature as in this comment https://github.com/bids-standard/bids-specification/pull/820#discussion_r685956430, i.e. whether the link resolves directly (i.e. it is the actual location where the file is present) or indirectly (it is a location from which the dataset can be downloaded, and within that dataset the file is to be found).

Rather than writing "In case shipping the **data** that is pointed to is impossible, ...", or "In case shipping the **dataset** that is pointed to is impossible, ...", you could also write "In case shipping the **file** that is pointed to is impossible, ...". The phrase "shipping data" is less clear than the other two.